src/Resources/Products/Associations.ts

```diff
@@ -0,0 +1,26 @@
+import AbstractResource from '../AbstractResource';
+
+export default class Products extends AbstractResource {
+    initialise(): void {
+        this.resourcePath = '/products/associations';
+        this.defaultAccess = 'guest';
+    }
+
+    /**
+     * Get a single product association
+     * e.g. `/products/associations/disney`.
+     */
+    public get: (productAssociationSlug: string) => void = this.createMethodFromPartialSpec({
```

Comment:
This function can also take an axios config as it's second param, so that needs to be typed.

Finally, it shouldn't return void, so that should probably change to. 

We probably want a type of `HttpClientResponse` here 

```suggestion
    public get: (productAssociationSlug: string, ...args: HttpClientConfig) => HttpClientResponse = this.createMethodFromPartialSpec({
```